### PR TITLE
Print helpful error if try to use CLI commands outside of blitz app

### PR DESCRIPTION
**Pull request by Zeko369**
### Pull Request Type

- [x] Feature
- [x] Bug fix
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Tests related changes
- [ ] Other (please describe): Dependency updates

### What's the reason for the change? :question:

Closes: #189

### What are the changes and their implications? :gear:

Check if inside of blitz app

### Does this introduce a breaking change? :warning:

- [x] Yes
- [ ] No

### Other information

I made it so it checks you're in the root of the project, because although some commands can work throughout the project structure, some fail so we should encourage people to just run it from the root path

image

### CODE REVIEW

**Comment by Zeko369** [comment]
Closes #189

File: packages/cli/src/checkBeforeRunning.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const whitelistGlobal = ['new']
+
+const hook: Hook<'init'> = async function (options) {
+  if (options.id && whitelistGlobal.includes(options.id)) return
+
+  try {
+    await readFile('blitz.config.js')
+  } catch (err) {
+    this.error(`It appears that you're not running in a blitz project`)
+  }
```
Comment (flybayer) [inline packages/cli/src/checkBeforeRunning.ts:13]: Ok few things:

1. `blitz.config.js` is optional, so this won't work. We do need robust logic for detecting if we are in a blitz app. Maybe if `blitz` is a dependency of the package.json? 
2. Put the above detection logic in it's own file so we can reuse elsewhere
3. If not at root of a blitz project, check if we are in a sub directory of a blitz project and have a specific error message for this.
4. Else have an error about not being in a blitz project.

Comment (flybayer) [reply packages/cli/src/checkBeforeRunning.ts:13]: **Error message for in a sub-folder:** "You are currently in a sub-folder of your Blitz app, but this command must be used from the root of your project."

**Error message for not in a blitz app:** "You are not inside a Blitz project, so this command won't work. You can create a new app with `blitz new myapp` or see help with `blitz help`

**Comment by flybayer** [comment]
Sweeet! Two things:

1. Can you use the `pkg-dir` package that's already installed instead of `find-package-json`?
2. Could you add a detailed comment in the utils file describing the blitz app detection and all the possible outcomes?

Thank you so much!

**Comment by flybayer** [comment]
This is beautiful — great work!!!

**Comment by aem** [comment]
@Zeko369 i think we have commands that check for this manually, can you follow up and convert those usages to use the new utils?

**Comment by Zeko369** [comment]
@aem you pushed the new utils?

**Comment by aem** [comment]
Ah sorry my comment was a bit unclear. Since this hook/util you added handles it for all commands, can you remove any checks for `blitz.config` in existing commands?

**Pull request merged**